AIEnrichmentPipeline/functions/EnrichmentPipeline.Functions.Domain/Exceptions/ExistingUriException.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+namespace EnrichmentPipeline.Functions.Domain.Exceptions
+{
+    /// <summary>
+    /// Use when an Exception class is needed to indicate that the Uri is already in
+    /// use.
+    /// </summary>
+    public class ExistingUriException : Exception
```

Comment:
Just a note around usage of ApplicationException: the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.applicationexception) for that type suggest inheriting from System::Exception.

Reply:
Correct. But that isn't the point. By inheriting from ApplicationException you are stating that the exception you throw is intentional by your code while for example if you divide by 0 you will get an exception that inherit from Exception. 

If say your middleware wants, it can distinguish between the two and handle them differently.

Reply:
I take my advice back. Thanks.